### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by raguiar9080**
- Make wording a bit better and overall provide a bit more info when
possible.
- Add more commands on the build section and rework information present
there.
- Make publish section up to date with current code.
- Add some TODO.

Signed-off-by: Ruben Aguiar <r.aguiar9080@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,36 +63,50 @@ core library since it was quite good already.
...
+explicit there so not repeating it here.
 
 ## Build
 
 For a simple command that does all the building of the current source code inside docker as done on CI:
 `docker-compose -f docker-compose-ci.yml run build_from_schema`
 
-If you want to update from the remote schema API, you can do:
+Keep in mind the previous command does not incorporate any changes done to the `base_client` file.
+If you also want to build the final client (usefull if changes were made in `base_client`)update from the remote schema API, you can do:
```
Comment (rbruggem) [inline README.md:75]: * usefull -> useful
* add space after )

File: README.md [hunk 2]
```diff
@@ -61,36 +63,51 @@ core library since it was quite good already.
...
-Publishing currently requires manual intervention to up the version. The recommended way is to create a release in github. This will automatically be published to NPM once accepted.
+Publishing is currently done by creating a new release on Github. The current way is to:
+- Create a new git tag (v<number>).
+- Title of the release should be the release <number>.
+- Body of the release is free form, but you can try and use the auto generated one.
+
+Prior to publish step though, it is required to actually update the version as intended in the code.
+For this, use the `Makefile` commands `patch|minor|major-version` accordingly and commit the changes as usual.
 
 ## TODO
```
Comment (rbruggem) [inline README.md:107]: I'd have TODOs as Issues

File: README.md [hunk 3]
```diff
@@ -61,36 +63,51 @@ core library since it was quite good already.
...
+Also, you can check `README_DIST.md` for the README present on the npm package, example usage should be
+explicit there so not repeating it here.
 
 ## Build
 
 For a simple command that does all the building of the current source code inside docker as done on CI:
 `docker-compose -f docker-compose-ci.yml run build_from_schema`
 
-If you want to update from the remote schema API, you can do:
+Keep in mind the previous command does not incorporate any changes done to the `base_client` file.
```
Comment (rbruggem) [inline README.md:74]: incorporate -> include

File: README.md [hunk 4]
```diff
@@ -61,36 +63,51 @@ core library since it was quite good already.
...
+If you also want to build the final client (useful if changes were made in `base_client`) update from the
```
Comment (rbruggem) [inline README.md:75]: `final`: is the term final needed?

File: README.md [hunk 5]
```diff
@@ -61,36 +63,51 @@ core library since it was quite good already.
...
+remote schema API, you can do:
+`docker-compose -f docker-compose-ci.yml run rebuild_from_schema`
+
+If you want to update the current code based on the current remote openAPI schema, you can do:
```
Comment (rbruggem) [inline README.md:79]: Is `remote` schema the official schema at https://docs.lune.co/openapi.yml?

Comment (rbruggem) [reply README.md:79]: If so, I don't think the term remote is appropriate, because it's the default schema.

File: README.md (hunk 2 above)
Comment (raguiar9080) [reply README.md:107]: Hmm, issues seems like actual bugs. Especially with things like `add Rich types` don't seem like an issue. I'll rephrase to future work?

File: README.md (hunk 5 above)
Comment (raguiar9080) [reply README.md:79]: Well, it's still the `remote` schema. It just so happens it's also the default. I'll reword things here though, it actually reads a bit confusing. How about:
`If you want to rebuild the library based on the official openAPI schema (https://docs.lune.co/openapi.yml), you can do:`

File: README.md (hunk 2 above)
Comment (rbruggem) [reply README.md]: YES! perfect!

**Review by rbruggem — APPROVED** [review]

**Pull request merged**